pcsx2/System.h

```diff
@@ -42,41 +42,43 @@ class RecompiledCodeReserve;
 
 namespace HostMemoryMap
 {
-	static const u32 Size = 0x28000000;
+	static constexpr u32 _128mbOffset = 0x6000000;
+
+	static constexpr u32 Size = 0x28000000 + _128mbOffset; // 160mb(base) + 96MB(extra needed to bring up to 128MB)
 
 	// The actual addresses may not be equivalent to Base + Offset in the event that allocation at Base failed
 	// Each of these offsets has a debugger-accessible equivalent variable without the Offset suffix that will hold the actual address (not here because we don't want code using it)
 
 	// PS2 main memory, SPR, and ROMs
-	static const u32 EEmemOffset   = 0x00000000;
+	static constexpr u32 EEmemOffset   = 0x00000000; // 128mb
 
 	// IOP main memory and ROMs
-	static const u32 IOPmemOffset  = 0x04000000;
+	static constexpr u32 IOPmemOffset  = 0x04000000 + _128mbOffset;
```

Comment:
This will also break anyone's cheat tables if modifying anything in IOP memory, regardless of whether the option is enabled or not.

Reply:
I don't see any difference in addresses between pcsx2 builds with and without 128mb:
![image](https://user-images.githubusercontent.com/4904157/160776241-0073b81d-bfd0-4ebb-b962-1e3ad2df0058.png)
Seems like an unnecessary complication?

Reply:
I'm not talking about the guest addresses. I'm talking about the offset to IOP memory in the main PCSX2 memory map, which is often used by people with programs like cheat engine.

Reply:
No. People like to use the base pointer to the memory map (or EEmem), and then offset that. They should use the IOPmem pointer instead, but plenty of modders seem to rely on fixed addresses from the start of the memory map.

Reply:
> That's why I suggested relocating the additional memory to the end of the memory map, keeping the start of IOP memory the same.

I don't quite see how to reflect that in the code, mind pushing this change?

Reply:
Add an additional space between the end of IOP memory and the recompiler code area, and special case the physical EE pages above 32MB in VTLB when setting up the page pointers.

Reply:
Wait, but wouldn't that mean the offset of that memory will be different then?
I need `EEMainMemoryStart + offset` external writings too.

Reply:
Plus it's not the first time memory layout changes, I don't see why some abstract cheat engine users that use EE Mem to access IOP should be thought about here. Isn't it much better to provide an API to access the addresses of these memory regions?

Reply:
> Wait, but wouldn't that mean the offset of that memory will be different then?
> I need EEMainMemoryStart + offset external writings too.

Yes, that's what I was referring to (but it's only relevant for IOP memory, because the EE memory is still 0-32MB).

> Isn't it much better to provide an API to access the addresses of these memory regions?

There is one. You can use IPC, or the `EEmem` and `IOPmem` pointers. But they don't.

Reply:
> But they don't.

Then they can PR this change if it's necessary, I'd rather keep the memory whole, better for external access and pattern matching and whatnot.

Reply:
So you're not willing to address any of the concerns I've raised? (race conditions, live configuration changes, resetting, per-game configuration, memory map changes), despite me suggesting multiple methods of how it could be solved?

If not, then this discussion is pointless, and I'm removing myself/won't reply any further.

Reply:
I'm sorry, I just don't enough about the subject and don't understand some concerns 
- like, per-game configuration, memory is already allocated on pcsx2 startup, what's there related specifically to the game?
- memory map changes, "fixing" iop pointer that is used by 3rd party, but making EE memory two regions, requiring to find two pointers now? Remember that almost no games will use that memory and the whole point of this PR is to make it available for external tools, which then may result to more pull requests to pcsx2 which utilize it.
My solution is to remove the toggle and have more memory always.
Would be great if you could help code wise in a different pull request or this one with your vision of how the code should look like. Because I only need more memory, not this concrete implementation.

Reply:
Race condition: accessing GameLoaded/GameStarted on the UI thread is unsafe. Because it can change between opening the settings dialog and when the user interacts with it.

Per-game configuration: The Qt interface lets you set near any setting in a per-game profile, and those settings are applied when the ELF is loaded. Which in your setting's case, means that the 128MB flag won't reflect the TLB virtual state.

Live configuration changes: Same deal, except changing the setting while the game is running, which you didn't address in Qt. The "safe" way to do this, like I said, would be to reinitialize both the physical and virtual VTLB mappings when the 128MB flag changes. But then you risk messing with any mappings the game happened to put in this area.

Save states: 
1. You're breaking save state compatibility (which isn't that huge of a deal, but when it happens, users tend to get annoyed.
2. What happens if someone saves a state with 128MB enabled, and loads with it disabled? Or vice versa?

Memory map changes: As I've stated repeatedly now, you're moving the start of IOP further up the memory map, and that affects anyone with hardcoded offsets anywhere above 32MB.

Furthermore, perhaps more concerningly, this affects the physical side of the VTLB mapping; you're unconditionally physically mapping 128MB instead of 32MB, which means that the upper 96MB will be accessible to games regardless instead of raising an exception. I'm not aware of anything which necessarily depends on this, but it does mean debugging issues may be more difficult, as code "running away" due to a bug will take much longer to TLB miss, if at all. See https://github.com/PCSX2/pcsx2/blob/master/pcsx2/Memory.cpp#L152

Reply:
> Race condition: accessing GameLoaded/GameStarted on the UI thread is unsafe. Because it can change between opening the settings dialog and when the user interacts with it.

I reverted that change. Let's remove the toggle next.

> Per-game configuration: The Qt interface lets you set near any setting in a per-game profile, and those settings are applied when the ELF is loaded. Which in your setting's case, means that the 128MB flag won't reflect the TLB virtual state.

> Live configuration changes: Same deal, except changing the setting while the game is running, which you didn't address in Qt. The "safe" way to do this, like I said, would be to reinitialize both the physical and virtual VTLB mappings when the 128MB flag changes. But then you risk messing with any mappings the game happened to put in this area.

Let's remove the toggle.

> Save states:

> You're breaking save state compatibility (which isn't that huge of a deal, but when it happens, users tend to get annoyed.
> What happens if someone saves a state with 128MB enabled, and loads with it disabled? Or vice versa?

Let's remove the toggle. Plus, in my opinion, the possibilities that more memory brings well worth some broken save states.

> Memory map changes: As I've stated repeatedly now, you're moving the start of IOP further up the memory map, and that affects anyone with hardcoded offsets anywhere above 32MB.
I have hardcoded offsets right now too that will break if you do this change preserving IOP offset. I have zero clue why would anyone need IOP. But when I used EE mem offset that was hardcoded to 0x20000000 in x86 pcsx2, nobody notified me when it changed. Then it changed again in x64 version, nobody told me either. From my perspective, things like this happen all the time. I understand your explanation, but don't understand how to implement it, so I won't.

> Furthermore, perhaps more concerningly, this affects the physical side of the VTLB mapping; you're unconditionally physically mapping 128MB instead of 32MB, which means that the upper 96MB will be accessible to games regardless instead of raising an exception. I'm not aware of anything which necessarily depends on this, but it does mean debugging issues may be more difficult, as code "running away" due to a bug will take much longer to TLB miss, if at all. See https://github.com/PCSX2/pcsx2/blob/master/pcsx2/Memory.cpp#L152

¯\_(ツ)_/¯

Reply:
> I understand your explanation, but don't understand how to implement it, so I won't.
Then learn

Reply:
> > I understand your explanation, but don't understand how to implement it, so I won't.
> > Then learn

I prefer the way it is now. Complicating code to preserve a pointer that nobody should even access the way they allegedly do now, is not worth the effort for me, sorry.